M365/src/DeleteContactsFromFolder/DeleteContactsFromFolder.ps1

```diff
@@ -0,0 +1,188 @@
+                    $contacts.AddRange($contactsResponse.Content.value)
+                    $query = $contactsResponse.Content.'@odata.nextLink'.replace("https://graph.microsoft.com/v1.0/", "")
+                    $listContactsInFolder.Query = $query
+                    $contactsResponse = Invoke-GraphApiRequest @listContactsInFolder
+                } until (-not($contactsResponse.Content.'@odata.nextLink'))
+
+                Write-Host "Number of contacts in the folder: '$($folderObj.displayName)' is: $($contacts.Count)"
+                $userConfirmation = Read-Host -Prompt "About to delete the folder and its content. Are you sure Y/N"
+
+                if ($userConfirmation.ToLower() -eq "y") {
+                    Write-Host "Deleting Folder contents..." -ForegroundColor Cyan
+                    # Loop through and delete the contacts
+                    foreach ($contact in $contacts) {
+                        Write-Host "Now processing: '$($contact.displayName)'" -ForegroundColor Cyan
+                        Write-Host "Id:  $($contact.id)" -ForegroundColor Cyan
+                        $deleteContactParams = @{
+                            Query              = "me/contactFolders/$($folderObj.id)/contacts/$($contact.id)"
+                            AccessToken        = $token.AccessToken
+                            Method             = "DELETE"
+                            ExpectedStatusCode = 204
+                            GraphApiUrl        = $graphApiUrl
+                        }
+                        $deleteContactResponse = Invoke-GraphApiRequest @deleteContactParams
+                        if ($deleteContactResponse.Successful -eq $false) {
+                            Write-Host "Unable to delete the contact" -ForegroundColor Red
+                            $unableToDeleteAllContacts = $true
+                        } else {
+                            Write-Host "Deleted contact successfully" -ForegroundColor Green
+                        }
+                    }
+
+                    if ($unableToDeleteAllContacts -eq $false) {
+                        Write-Host "Deleted folder content, now deleting the folder itself..." -ForegroundColor Cyan
+                        # Remove the folder itself
+                        $deleteContactFolderParams = @{
+                            Query              = "me/contactFolders/$($folderObj.id)"
+                            AccessToken        = $token.AccessToken
+                            Method             = "DELETE"
+                            ExpectedStatusCode = 204
+                            GraphApiUrl        = $graphApiUrl
+                        }
+                        $deleteContactFolderResponse = Invoke-GraphApiRequest @deleteContactFolderParams
+                        if ($deleteContactFolderResponse.Successful -eq $false) {
+                            Write-Host "Unable to delete the folder: '$($folderObj.displayName)'" -ForegroundColor Red
```

Comment:
Again, doesn't provide context to the issue making it hard to debug without having someone walk through the steps manually